app/lib/zif/ui/input.rb

```diff
@@ -0,0 +1,76 @@
+module Zif
+  module UI
+    # A basic input field that can accept keystrokes and record and display changes
+    # to have it recieve key strokes it needs to be registered
+    #    $game.services[:input_service].register_key_pressable(@input)
+    #
+    # to be able to update the display in response to those, you should to add it as a label every tick
+    #    perform tick args
+    #      ...
+    #      $gtk.args.outputs.labels << [@input]
+    #      ...
+    #
+    class Input < Label
+      include Zif::KeyPressable
+      include Zif::Serializable
+
+      attr_rect
+      attr_accessor :max_length
+      attr_accessor :desired_keys, :special_keys, :map_keys
+      attr_accessor :has_focus
```

Comment:
These need to be split into their own lines and documented, see `clickable.rb` for an example

Additionally each method here should probably be marked `# @api private`